helmfile/charts/prometheus-alerts/templates/rules/fluentd.yaml

```diff
@@ -0,0 +1,86 @@
+{{- if and .Values.defaultRules.create .Values.defaultRules.rules.fluentd }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ printf "%s-%s" (include "prometheus-alerts.fullname" .) "fluentd" | trunc 63 | trimSuffix "-" }}
+  labels:
+    app: {{ template "prometheus-alerts.fullname" . }}
+{{ include "prometheus-alerts.labels" . | indent 4 }}
+{{- if .Values.defaultRules.labels }}
+{{ toYaml .Values.defaultRules.labels | indent 4 }}
+{{- end }}
+{{- if .Values.defaultRules.annotations }}
+  annotations:
+{{ toYaml .Values.defaultRules.annotations | indent 4 }}
+{{- end }}
+spec:
+  groups:
+  - name: fluentd
+    rules:
+    - alert: FluentdNodeDown
+      expr: up{job=~"fluentd.*"} == 0
+      for: 10m
+      labels:
+        service: fluentd
+        severity: warning
+      annotations:
+        summary: fluentd cannot be scraped
+        description: Prometheus could not scrape {{ "{{ $labels.job }}" }} for more than 10 minutes
+    - alert: FluentdNodeDown
+      expr: up{job=~"fluentd.*"} == 0
+      for: 30m
+      labels:
+        service: fluentd
+        severity: critical
+      annotations:
+        summary: fluentd cannot be scraped
+        description: Prometheus could not scrape {{ "{{ $labels.job }}" }} for more than 30 minutes
+    - alert: FluentdQueueLength
+      expr: rate(fluentd_status_buffer_queue_length[5m]) > 0.3
+      for: 1m
+      labels:
+        service: fluentd
+        severity: warning
+      annotations:
+        summary: fluentd node are failing
+        description: In the last 5 minutes, fluentd queues increased 30%. Current value is {{ "{{ $value }}" }}
+    - alert: FluentdQueueLength
+      expr: rate(fluentd_status_buffer_queue_length[5m]) > 0.5
+      for: 1m
+      labels:
+        service: fluentd
+        severity: critical
+      annotations:
+        summary: fluentd node are critical
+        description: In the last 5 minutes, fluentd queues increased 50%. Current value is {{ "{{ $value }}" }}
+    - alert: FluentdRecordsCountsHigh
+      expr: >
+        sum(rate(fluentd_output_status_emit_records{job="fluentd-metrics"}[5m]))
+        BY (instance) >  (3 * sum(rate(fluentd_output_status_emit_records{job="fluentd-metrics"}[15m]))
+        BY (instance))
+      for: 1m
+      labels:
+        service: fluentd
+        severity: critical
+      annotations:
+        summary: fluentd records count are critical
+        description: In the last 5m, records counts increased 3 times, comparing to the latest 15 min.
```

Comment:
How stable are these metric? I'm slightly afraid that since these are critical they might give us a lot of alarms if this is not very stable.

Reply:
FluentdRecordsCountsHigh seems that will be triggered 2-3 times/day
I can change the severity or the evaluation time, but this will deviate from the upstream so might be overwritten in the future

Reply:
Do you know if it was an actual problem for us as well 2-3 times per day or if most of them were smaller issues that we could have ignored?
I agree that we would prefer not to deviate from upstream. But we do not want to get woken up every night because of this.

Reply:
Those sounds like good changes. I vote for 10m evaluations.

Reply:
done